apps/astarte_appengine_api/lib/astarte_appengine_api/device/queries.ex

```diff
@@ -285,19 +284,66 @@ defmodule Astarte.AppEngine.API.Device.Queries do
         endpoint_id,
         path,
         value_timestamp,
-        reception_timestamp
+        reception_timestamp,
+        opts
       ) do
-    property_table = String.replace(interface_descriptor.storage, "datastreams", "properties")
+    insert_path(
+      db_client,
+      device_id,
+      interface_descriptor,
+      endpoint_id,
+      path,
+      value_timestamp,
+      reception_timestamp,
+      opts
+    )
+  end
 
+  def insert_path_into_db(
+        db_client,
+        device_id,
+        %InterfaceDescriptor{storage_type: :one_object_datastream_dbtable} = interface_descriptor,
+        endpoint_id,
+        path,
+        value_timestamp,
+        reception_timestamp,
+        opts
+      ) do
+    insert_path(
+      db_client,
+      device_id,
+      interface_descriptor,
+      endpoint_id,
+      path,
+      value_timestamp,
+      reception_timestamp,
+      opts
+    )
+  end
+
+  defp insert_path(
```

Comment:
Instead of adding this, match above with
```elixir
...
 %InterfaceDescriptor{storage_type: storage_type
...
when storage_type in  [:one_object_datastream_dbtable, :multi_interface_individual_datastream_dbtable]
```